backend/internal/database/setting.go

```diff
@@ -10,6 +10,11 @@ import (
 // the value stored for a setting, given the setting's name.  It returns
 // an error for illegal settings.
 func getSetting(settingName string) (int, string, error) {
+	err := db.Ping()
+	if err != nil {
+		return 0, "", fmt.Errorf("db.Ping() failed: %w", err)
+	}
```

Comment:
With these lines we can detect if a broken connection has happened. For example, when the database goes down or is unavailable.